lib/Models/WebFeatureServiceCapabilities.ts

```diff
@@ -0,0 +1,171 @@
+  static fromUrl: (
+    url: string
+  ) => Promise<WebFeatureServiceCapabilities> = createTransformer(
+    (url: string) => {
+      return makeRealPromise(loadXML(url)).then(function(capabilitiesXml: any) {
+        const json = xml2json(capabilitiesXml);
+        if (!defined(json.ServiceIdentification)) {
+          throw new TerriaError({
+            title: "Invalid GetCapabilities",
+            message:
+              `The URL ${url} was retrieved successfully but it does not appear to be a valid Web Feature Service (WFS) GetCapabilities document.` +
+              `\n\nEither the catalog file has been set up incorrectly, or the server address has changed.`
+          });
+        }
+
+        return new WebFeatureServiceCapabilities(capabilitiesXml, json);
+      });
+    }
+  );
+
+  /**
+   * Get CapabilitiesService (in WMS form)
+   */
+  get Service(): CapabilitiesService {
+    const serviceProviderJson = this.json["ServiceProvider"];
+    const serviceIdentificationJson = this.json["ServiceIdentification"];
+    const serviceAddressJson =
+      serviceProviderJson?.["ServiceContact"]?.["ContactInfo"]?.["Address"];
+    const service: CapabilitiesService = {
+      Title: serviceIdentificationJson?.["Title"],
+      Abstract: serviceIdentificationJson?.["Abstract"],
+      Fees: serviceIdentificationJson?.["Fees"],
+      AccessConstraints: serviceIdentificationJson?.["AccessConstraints"],
+      KeywordList: {
+        Keyword: serviceIdentificationJson?.["Keywords"]?.["Keyword"]
+      },
+      ContactInformation: {
+        ContactPersonPrimary: {
+          ContactPerson:
+            serviceProviderJson?.["ServiceContact"]?.["IndividualName"],
+          ContactOrganization: serviceProviderJson?.["ProviderName"]
+        },
+        ContactPosition:
+          serviceProviderJson?.["ServiceContact"]?.["PositionName"],
+        ContactAddress: {
+          Address: serviceAddressJson?.["DeliveryPoint"],
+          City: serviceAddressJson?.["City"],
+          StateOrProvince: serviceAddressJson?.["AdministrativeArea"],
+          PostCode: serviceAddressJson?.["PostalCode"],
+          Country: serviceAddressJson?.["Country"]
+        },
+        ContactVoiceTelephone:
+          serviceProviderJson?.["ServiceContact"]?.["ContactInfo"]?.["Phone"]?.[
+            "Voice"
+          ],
+        ContactFacsimileTelephone:
+          serviceProviderJson?.["ServiceContact"]?.["ContactInfo"]?.["Phone"]?.[
+            "Facsimile"
+          ],
+        ContactElectronicMailAddress:
+          serviceProviderJson?.["ServiceContact"]?.["ContactInfo"]?.[
+            "Address"
+          ]?.["ElectronicMailAddress"]
+      }
+    };
+    return service;
+  }
+
+  get featureTypes(): FeatureType[] {
+    const featureTypesJson = this.json.FeatureTypeList?.FeatureType as Array<
+      any
+    >;
+    if (!isDefined(featureTypesJson) || !Array.isArray(featureTypesJson)) {
+      return [];
+    }
```

Comment:
This will cause the server with only one layer to fail (say the group is empty). It is pretty common to have such services when using isolated workspace in geoserver.

Reply:
Good pickup @zoran995 

For example this WFS URL
 https://data.gov.au/geoserver/australia-s-indigenous-forest-estate-2018/wfs

Reply:
Thanks @zoran995 , all fixed!

Reply:
@rowanwins that URL doesn't have any <FeatureType>, but I tested with my own Geoserver and it seems to work fine

Reply:
Hmm weird, looks like the data.gov.au WFS thing might be broken...